core/src/test/java/org/elasticsearch/action/search/MultiSearchRequestTests.java

```diff
@@ -163,10 +169,30 @@ public void testResponseErrorToXContent() throws IOException {
         MultiSearchResponse response = new MultiSearchResponse(new MultiSearchResponse.Item[]{new MultiSearchResponse.Item(null, new IllegalStateException("foobar")), new MultiSearchResponse.Item(null, new IllegalStateException("baaaaaazzzz"))});
         XContentBuilder builder = XContentFactory.jsonBuilder();
         response.toXContent(builder, ToXContent.EMPTY_PARAMS);
-        assertEquals("\"responses\"[{\"error\":{\"root_cause\":[{\"type\":\"illegal_state_exception\",\"reason\":\"foobar\"}],\"type\":\"illegal_state_exception\",\"reason\":\"foobar\"}},{\"error\":{\"root_cause\":[{\"type\":\"illegal_state_exception\",\"reason\":\"baaaaaazzzz\"}],\"type\":\"illegal_state_exception\",\"reason\":\"baaaaaazzzz\"}}]",
+        assertEquals("\"responses\"[{\"error\":{\"root_cause\":[{\"type\":\"illegal_state_exception\",\"reason\":\"foobar\"}],\"type\":\"illegal_state_exception\",\"reason\":\"foobar\"},\"status\":500},{\"error\":{\"root_cause\":[{\"type\":\"illegal_state_exception\",\"reason\":\"baaaaaazzzz\"}],\"type\":\"illegal_state_exception\",\"reason\":\"baaaaaazzzz\"},\"status\":500}]",
                 builder.string());
     }
 
+    public void testResponseSuccessToXContent() throws IOException {
+        MultiSearchResponse response = fakeMultiSearchResponse();
+        XContentBuilder builder = XContentFactory.jsonBuilder();
+        response.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        assertEquals("\"responses\"[{\"took\":1,\"timed_out\":false,\"terminated_early\":false,\"_shards\":{\"total\":1,\"successful\":1,\"failed\":0},\"hits\":{\"total\":1,\"max_score\":0.0,\"hits\":[{\"_type\":\"type\",\"_id\":\"id\",\"_score\":\"-Infinity\",\"fields\":{\"field\":[\"somevalue\"]}}]},\"status\":200},{\"took\":1,\"timed_out\":false,\"terminated_early\":false,\"_shards\":{\"total\":1,\"successful\":1,\"failed\":0},\"hits\":{\"total\":1,\"max_score\":0.0,\"hits\":[{\"_type\":\"type\",\"_id\":\"id\",\"_score\":\"-Infinity\",\"fields\":{\"field\":[\"somevalue\"]}}]},\"status\":200}]",
+            builder.string());
+    }
+
+    private MultiSearchResponse fakeMultiSearchResponse(){
```

Comment:
You can merge the body of this method directly in `testResponseSuccessToXContent()` since it is not used anywhere else.

Also, we noticed in #18765 that `msearch` has no RESt tests. So I think it would be better to remove this test method and add REST tests like in `rest-api-spec/src/main/resources/rest-api-spec/test/msearch/10_basic.yaml` that check for `status`.